tests/pubsub/testgossipinternal.nim

```diff
@@ -57,6 +57,10 @@ suite "GossipSub internal":
       let topic = "foobar"
       gossipSub.gossipsub[topic] = initHashSet[string]()
 
+      # our implementation requires that topic is in gossipSub.topics
+      # for this test to work properly and publish properly
+      gossipSub.topics[topic] = Topic()
```

Comment:
This shouldn't be the case for fanout, in fact it is the opposite, the topic should be absent from the topics table. We should probably even add a check `check: topic notin gossipSub.topics`.

Reply:
Well the implementation indeed requires this :) it was working before because tests were too quick and did not heartbeat and clear gossip table
I get it tho.. will try to fix it

Reply:
fixed/removed currently